core/coreapi/interface/tests/key.go

```diff
@@ -125,11 +154,12 @@ func TestGenerateSize(t *testing.T) {
 	}
 }
 
-func TestGenerateType(t *testing.T) {
-	ctx := context.Background()
+func (tp *provider) TestGenerateType(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
These could go below skip?

Reply:
(yes but I didn't see that comment before I merged... oh well, it doesn't really matter much).